app/screens/channel_files/header.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useCallback, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {Text, View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import Badge from '@components/badge';
+import CompassIcon from '@components/compass_icon';
+import Filter, {DIVIDERS_HEIGHT, FILTER_ITEM_HEIGHT, FilterData, NUMBER_FILTER_ITEMS} from '@components/files/file_filter';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {TITLE_SEPARATOR_MARGIN, TITLE_SEPARATOR_MARGIN_TABLET, TITLE_HEIGHT} from '@screens/bottom_sheet/content';
+import {bottomSheet} from '@screens/navigation';
+import {FileFilter, FileFilters} from '@utils/file';
+import {bottomSheetSnapPoint} from '@utils/helpers';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const TEST_ID = 'channel_files';
+
+type Props = {
+    onFilterChanged: (filter: FileFilter) => void;
+    selectedFilter: FileFilter;
+}
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flexDirection: 'row',
+            justifyContent: 'space-between',
+            paddingHorizontal: 20,
+            paddingTop: 20,
+            backgroundColor: theme.centerChannelBg,
+        },
+        title: {
+            color: theme.centerChannelColor,
+            ...typography('Heading', 300, 'SemiBold'),
+        },
+        badge: {
+            backgroundColor: theme.buttonBg,
+            borderColor: theme.centerChannelBg,
+            marginTop: 2,
+        },
+        iconsContainer: {
+            alignItems: 'center',
+            flexDirection: 'row',
+            marginLeft: 'auto',
+        },
+    };
+});
+
+const Header = ({
+    onFilterChanged,
+    selectedFilter,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+    const intl = useIntl();
+    const {bottom} = useSafeAreaInsets();
+    const isTablet = useIsTablet();
+    const hasFilters = selectedFilter !== FileFilters.ALL;
+
+    let messagesText = intl.formatMessage({id: 'screen.channel_files.header.recent_files', defaultMessage: 'Recent Files'});
+    if (hasFilters) {
+        messagesText = intl.formatMessage({id: FilterData[selectedFilter].id, defaultMessage: FilterData[selectedFilter].defaultMessage});
+    }
```

Comment:
the check can be done differently to avoid one computation here.

```suggestion
    let messagesText = hasFilters : intl.formatMessage({id: FilterData[selectedFilter].id, defaultMessage: FilterData[selectedFilter].defaultMessage}); ? intl.formatMessage({id: 'screen.channel_files.header.recent_files', defaultMessage: 'Recent Files'});
```

the above suggestion properly formatted.